drivers/led_strip/apa102.c

```diff
@@ -21,7 +21,18 @@ static int apa102_update(const struct device *dev, void *buf, size_t size)
 {
 	const struct apa102_config *config = dev->config;
 	static const uint8_t zeros[] = { 0, 0, 0, 0 };
-	static const uint8_t ones[] = { 0xFF, 0xFF, 0xFF, 0xFF };
+
+	/*
+	 * the only function of the “End frame” is to supply more clock pulses
+	 * to the string until the data has permeated to the last LED. The
+	 * number of clock pulses required is exactly half the total number
+	 * of LEDs in the string
+	 */
+	BUILD_ASSERT(sizeof(struct led_rgb) == 4);
```

Comment:
This is not a problem, but it is customary not to write `BUILD_ASSERT` inside a function.
I think it would be better to put it inside `APA102_DEVICE` macro.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/76533 split into another PR to remove how brightness is being used